homeassistant/components/pvpc_hourly_pricing/__init__.py

```diff
@@ -74,12 +57,16 @@ class ElecPricesDataUpdateCoordinator(DataUpdateCoordinator[EsiosApiData]):
 
     def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
         """Initialize."""
+        api_token = None
+        if entry.data.get(CONF_USE_API_TOKEN, False):
```

Comment:
>  If `CONF_API_TOKEN` is `None` then you know the API token is not in use. 

Hi Tom, thanks for the review 🍻 

**why the bool?**, right?

It didn't look pretty to me as well when I did the changes 😢, but I didn't find a better way of defining an **optional** API Token with the current UI.

Without the bool, AFAIK, there is no way to remove/reset/disable the token field in the config flow: once it's shown, it has to be filled, right? And then `CONF_API_TOKEN` won't be `None` ever again, so the user cannot toggle it back unless fully removing the integration to setup it again, which is not ideal, I thought.

**Do you know a better way to do this?** 🙏

Should I use the boolean flag just for the config-flow steps (to show the checkbox for the user), and then unset the token value internally if not checked, to store only the optional token in the config-entry?

Reply:
> Should I use the boolean flag just for the config-flow steps (to show the checkbox for the user), and then unset the token value internally if not checked, to store only the optional token in the config-entry?

I'm going to push that change, to avoid storage of `CONF_USE_API_TOKEN` in `entry.data` and `entry.options`, but maintaining it for the config wizard, ok?

Reply:
Yes, that is what I meant. So resolved.